apps/client/src/app/pages/inventory-optimizer/optimizations/not-used-in-lists.ts

```diff
@@ -0,0 +1,42 @@
+import { InventoryOptimizer } from './inventory-optimizer';
+import { InventoryItem } from '../../../model/user/inventory/inventory-item';
+import { UserInventory } from '../../../model/user/inventory/user-inventory';
+import { ListRow } from '../../../modules/list/model/list-row';
+import { LazyDataService } from '../../../core/data/lazy-data.service';
+import { ListsFacade } from '../../../modules/list/+state/lists.facade';
+import { Memoized } from '../../../core/decorators/memoized';
+import { OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs/Subscription';
+import * as _ from 'lodash';
+
+export class NotUsedInLists extends InventoryOptimizer implements OnDestroy {
+  allMaterialsOnListsSubscription: Subscription;
+  allMaterialsOnLists: Set<number> = new Set();
+
+  constructor(private lazyData: LazyDataService, private listsFacade: ListsFacade) {
+    super();
+
+    this.allMaterialsOnListsSubscription = listsFacade.allListDetails$.subscribe(lists => {
+      this.allMaterialsOnLists = new Set(_.flatMap(lists, list => list.items.map(item => item.id)));
```

Comment:
Instead of `flatMap`, I'd use `[].concat.apply([], lists.map(list => list.items.map(item => item.id)));`